### PR TITLE
DevTools: Improve browser extension iframe support

### PR DESCRIPTION
**Pull request by omarsy**
## Summary
This pull requests is intended to improve iframe support for devtools browser extensions(#18945), without add a code to  pass the hook through from the parent.

image

### CODE REVIEW

**Review by hellbound1337 — APPROVED** [review]

File: packages/react-devtools-shared/src/hook.js [hunk 1]
```diff
@@ -21,6 +21,25 @@ export function installHook(target: any): DevToolsHook | null {
...
 
+  function getMainWindow(targetWindow: any): any {
+    if (!canAccessParentWindow(targetWindow) || isMainWindow(targetWindow)) {
+      return targetWindow;
+    }
+    return getMainWindow(targetWindow.parent);
+  }
+
+  function isMainWindow(targetWindow: any): boolean {
+    return targetWindow.self === targetWindow.top;
```
Comment (bvaughn) [inline packages/react-devtools-shared/src/hook.js:32]: Why `window.self` rather than just `window`?

**Review by bvaughn — APPROVED** [review]
Thanks! This looks good.

File: packages/react-devtools-shared/src/hook.js (hunk 1 above)
Comment (omarsy) [reply packages/react-devtools-shared/src/hook.js:32]: It is the same but `self` can work with worker(it is more flexible)

**Comment by omarsy** [comment]
Thank you @bvaughn !

File: packages/react-devtools-shared/src/hook.js (hunk 1 above)
Comment (bvaughn) [reply packages/react-devtools-shared/src/hook.js:32]: Ok just wanted to confirm I wasn't missing anything. 😄

**Comment by bvaughn** [comment]
Looks like this change broke the development test harness. To repro:
1. Disable the React DevTools extension entirely.
2. Run the [local test harness](https://github.com/facebook/react/tree/master/packages/react-devtools-shell).
3. See nothing in the Components tab.

I think I'm going to have to revert it for now. Any interest in looking back into it @omarsy?

**Comment by omarsy** [comment]
ok @bvaughn  I'm looking it

**Pull request merged**